### PR TITLE
refactor: add output dimensionality as a parameter in AbstractInterpolations

### PR DESCRIPTION
**Pull request by sathvikbhagavan**
fixes: #328

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -42,7 +42,7 @@ StableRNGs = "1"
 Symbolics = "5.29, 6"
 Test = "1"
 Zygote = "0.6.70"
-julia = "1.10"
+julia = "1.11"
```
Comment (ChrisRackauckas) [inline Project.toml:45]:
```suggestion
julia = "1.10"
```

no we stay on lts

File: Project.toml [hunk 2]
```diff
@@ -30,7 +30,7 @@ ChainRulesCore = "1.24"
 FindFirstFunctions = "1.3"
 FiniteDifferences = "0.12.31"
 ForwardDiff = "0.10.36"
-LinearAlgebra = "1.10"
+LinearAlgebra = "1.11"
```
Comment (ChrisRackauckas) [inline Project.toml:33]: why is this required?

File: Project.toml (hunk 1 above)
Comment (sathvikbhagavan) [reply Project.toml]: Alright, I have reverted the commit.

File: Project.toml (hunk 2 above)
Comment (sathvikbhagavan) [reply Project.toml]: I have reverted the commit.

File: src/interpolation_utils.jl [hunk 3]
```diff
@@ -59,6 +59,19 @@ function spline_coefficients!(N, d, k, u::AbstractVector)
...
+    return (1,)
+end
+
+function get_output_dim(u::AbstractVector)
+    return (length(first(u)),)
+end
+
+function get_output_dim(u::AbstractArray)
+    return size(u)[1:(end - 1)]
+end
```
Comment (devmotion) [inline src/interpolation_utils.jl:73]: @sathvikbhagavan @ChrisRackauckas This breaks type inference of the constructors which causes problems in my downstream package (see #389). One way around this would be to allow (require?) users to specify the dimension as a type parameter when constructing the interpolation method.

However, it's not clear to me why this type parameter should be included at all. Arguably it also doesn't make sense at all for nonnumeric types (that I actually encounter when working with `ConstantInterpolation`). Would it be OK to revert this change?

If the output dimension is actually of interest to someone then it seems easy enough for them to use the logic of `get_output_dim` directly. Possibly this could also be made official by adding something like `output_dim` to the API (if actually necessary). This could also e.g. error for non-numeric types.
